addons/auth_signup/models/res_users.py

```diff
@@ -76,16 +76,18 @@ def signup(self, values, token=None):
 
     @api.model
     def _signup_create_user(self, values):
-        """ create a new user from the template user """
-        get_param = self.env['ir.config_parameter'].sudo().get_param
-        template_user_id = literal_eval(get_param('auth_signup.template_user_id', 'False'))
-        template_user = self.browse(template_user_id)
-        assert template_user.exists(), 'Signup: invalid template user'
+        """ signup a new user using the template user """
 
         # check that uninvited users may sign up
         if 'partner_id' not in values:
-            if get_param('auth_signup.invitation_scope', 'b2b') != 'b2c':
+            if self.env['ir.config_parameter'].sudo().get_param('auth_signup.invitation_scope', 'b2b') != 'b2c':
                 raise SignupError(_('Signup is not allowed for uninvited users'))
+        return self._create_user_from_template(values)
+
+    def _create_user_from_template(self, values):
+        template_user_id = literal_eval(self.env['ir.config_parameter'].sudo().get_param('base.template_portal_user_id', 'False'))
+        template_user = self.browse(template_user_id)
+        assert template_user.exists(), 'Signup: invalid template user'
```

Comment:
Asserts are not a completely reliable way of raising exceptions as they may be dismissed depending on some options. I would instead perform an exists and raise a ValueError if not found.